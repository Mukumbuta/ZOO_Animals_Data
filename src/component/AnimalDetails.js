import React from 'react';
import { useSelector } from 'react-redux';
import DetailsPageHeader from './DetailsPageHeader';
import '../assets/detailsPageStyles.css';

const DisplayAnimalDetails = () => {
  const { animalsReducer } = useSelector((state) => state);

  let image;
  let animalName;
  let lifeSpan;
  let diet;
  let geoRange;
  let activeTime;
  let latinName;
  let maxLength;
  let minLength;
  let minWeight;
  let maxWeight;
  let habitat;

  // eslint-disable-next-line
  animalsReducer.map((animalia) => {
    // eslint-disable-next-line
    if (animalia.id == window.location.pathname.slice(1)) {
      image = animalia.image_link;
      animalName = animalia.name;
      lifeSpan = animalia.lifespan;
      diet = animalia.diet;
      geoRange = animalia.geo_range;
      activeTime = animalia.active_time;
      latinName = animalia.latin_name;
      maxLength = animalia.length_max;
      minLength = animalia.length_min;
      minWeight = animalia.weight_min;
      maxWeight = animalia.weight_max;
      habitat = animalia.habitat;
    }
  });

  return (
    <div>
      <div>
        <DetailsPageHeader name={animalName} />
      </div>
      <section className="animals-details-cont">
        <div id="details-header">
          <img src={image} id="species" alt="" />
          <h1>
            {animalName}
          </h1>
        </div>
        <ul className="details">
          <li>
            <h2>Latin Name</h2>
            <p>
              {latinName}
            </p>
          </li>
          <li>
            <h2>Life Span</h2>
            <p>
              {lifeSpan}
            </p>
          </li>
          <li>
            <h2>Habitat</h2>
            <p>
              {habitat}
            </p>
          </li>
          <li>
            <h2>Deit</h2>
            <p>
              {diet}
            </p>
          </li>
          <li>
            <h2>Active Time</h2>
            <p>
              {activeTime}
            </p>
          </li>
          <li>
            <h2>Geographical Range</h2>
            <p>
              {geoRange}
            </p>
          </li>

          <li>
            <h2>Weight</h2>
            <p>
              Maximum Weight:
              {' '}
              {maxWeight}
            </p>
            <p>
              Minimum Weight:
              {' '}
              {minWeight}
            </p>
          </li>

          <li>
            <h2>Length</h2>
            <p>
              Maximum Length:
              {' '}
              {maxLength}
            </p>
            <p>
              Minimum Length:
              {' '}
              {minLength}
            </p>
          </li>
        </ul>
      </section>
    </div>
  );
};
export default DisplayAnimalDetails;
