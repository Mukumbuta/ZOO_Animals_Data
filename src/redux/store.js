import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunkMiddleware from 'redux-thunk';
import animalsReducer from './animals/animals';

const reducer = combineReducers({
  animalsReducer,
});

const store = createStore(
  reducer,
  applyMiddleware(logger, thunkMiddleware),
);

export default store;
