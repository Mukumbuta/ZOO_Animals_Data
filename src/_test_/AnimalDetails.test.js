import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import AnimalDetails from '../component/AnimalDetails';

describe('Render AnimalDetails component', () => {
  test('render HomePage', () => {
    const tree = renderer
      .create(<Router><Provider store={store}><AnimalDetails /></Provider></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
