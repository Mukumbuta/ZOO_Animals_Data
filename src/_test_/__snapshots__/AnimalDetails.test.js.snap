// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render AnimalDetails component render HomePage 1`] = `
<div>
  <div>
    <header
      className="detailsPageHeader"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt=""
          className="header_img"
          src="accordion_back.png"
        />
      </a>
      <div
        className="mic_settings"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <img
            alt=""
            className="header_img settings"
            src="microphone.png"
          />
        </a>
        <a
          href="/"
          onClick={[Function]}
        >
          <img
            alt=""
            className="header_img"
            src="settings_icon.svg"
          />
        </a>
      </div>
    </header>
  </div>
  <section
    className="animals-details-cont"
  >
    <div
      id="details-header"
    >
      <img
        alt=""
        id="species"
      />
      <h1 />
    </div>
    <ul
      className="details"
    >
      <li>
        <h2>
          Latin Name
        </h2>
        <p />
      </li>
      <li>
        <h2>
          Life Span
        </h2>
        <p />
      </li>
      <li>
        <h2>
          Habitat
        </h2>
        <p />
      </li>
      <li>
        <h2>
          Deit
        </h2>
        <p />
      </li>
      <li>
        <h2>
          Active Time
        </h2>
        <p />
      </li>
      <li>
        <h2>
          Geographical Range
        </h2>
        <p />
      </li>
      <li>
        <h2>
          Weight
        </h2>
        <p>
          Maximum Weight:
           
        </p>
        <p>
          Minimum Weight:
           
        </p>
      </li>
      <li>
        <h2>
          Length
        </h2>
        <p>
          Maximum Length:
           
        </p>
        <p>
          Minimum Length:
           
        </p>
      </li>
    </ul>
  </section>
</div>
`;
